server/auth/gatekeeper.go

```diff
@@ -286,7 +286,12 @@ func (s *gatekeeper) getClientsForServiceAccount(ctx context.Context, claims *ty
 	if err != nil {
 		return nil, err
 	}
-	_, clients, err := s.clientForAuthorization(authorization)
+	var clients *servertypes.Clients
+	if s.restConfig != nil {
```

Comment:
is there a reason you only checked this condition in `getClientsForServiceAccount` and not in `getClients` as well?

I think effectively it will never be `nil` except in potentially a test case though

Reply:
Yes，it won't. Because a UT，maybe I can change the UT